File: Easy/118. Pascal's Triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<>(numRows);
        for (int i = 0; i < numRows; i++) {
            res.add(new ArrayList<>(i + 1));
            for (int j = 0; j < i + 1; j++) {
                if (j != 0 && j != i) {
                    res.get(i).add(res.get(i - 1).get(j - 1) + res.get(i - 1).get(j));
                } else {
                    res.get(i).add(1);
                }
            }
        }
        return res;
    }
}

File: Easy/118. Pascal's Triangle.py
from typing import List


class Solution:
    def generate(self, n: int) -> List[List[int]]:
        prev = [1]
        result = [prev]
        for i in range(1, n):
            curr = [1]
            for j in range(1, (i + 1) // 2):
                curr.append(prev[j - 1] + prev[j])
            curr += curr[::-1] if i % 2 else ([prev[i // 2] + prev[i // 2 - 1]] + curr[::-1])
            result.append(curr)
            prev = curr
        return result

class Solution1:
    def generate(self, n: int) -> List[List[int]]:
        result = [[1] for _ in range(n)]
        for i in range(1, n):
            curr = result[i]
            prev = result[i - 1]
            for j in range(1, len(prev)):
                curr.append(prev[j - 1] + prev[j])
            curr.append(1)
        return result